Update Releases Notes for ONVM v19.07 (#155)
sdnfv/openNetVM: docs/Releases.md

like `17.11` where the "major" number is the year and the "minor" number
is the month.

## v19.07 (7/19): NFD library and example NFs, minor improvements and bug fixes.
A CloudLab template is available with the latest release here: https://www.cloudlab.us/p/GWCloudLab/onvm

> **Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which will disable an expensive flow lookup for every incoming packet(instead packets are forwarded to the default service ID). This improves performance but flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.

Reviewer: ```suggestion
**Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which controls whether a flow lookup is performed for every incoming packet. If disabled, all packets are forwarded to the default service ID which improves performance. The flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.
```

Reviewer: If disabled, does it forward to the default service ID or the default service *chain*?

### NFD library with example NFS

> Add NFD, a C++-based NF developing compiler designed by Wenfei's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.

Reviewer: ```suggestion
Add example NFs based on NFD, a C++-based NF developing compiler designed by Wenfei Wu's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.
```

Reviewer: Does the code we merged actually include the NFD compiler? Or is it just some NFs that were produced by the NFD tool?  I think it is the latter, so we will need to adjust the text.

NFD compiles the NF logic into common C++ program by using table-form language to model NFs' behavior . With NFD's help, NF developer, free from the problems that porting NFs into specific network environment or deploying platforms, can easily concentrate on designing NF's core logic. 

A list of provided NFs using NFD library:
 - DNS Amplification Mitigation
 - Super Spread Detection
 - Heavy Hitter Detection
 - SYN Flood Detection
 - UDP Flood Detection
 - Stateless Firewall
 - Stateful Firewall
 - NAPT

> ### CI updates:

Reviewer: ```suggestion
### Continuous Integration updates:
```

CI got a few major updates this release:
 - CI will do basic lint checks and branch checks(all PRs should be submitted against the **develop** branch) for unathorized users
 - If CI is working on a request and recives another request it will append it to the queue instead of dropping it
 - CI will now run Pktgen as an additional test metric.

### Minor improvements

>  - Shared core functionality for messages

Reviewer: What does this mean exactly? This has a new API, right? We should explain it.  We used to have a section in the release notes about new/changed APIs.

>  - NF core reassignment on termination

Reviewer: Is reassignment enabled by default? Do we explain it in more detail anywhere else?  I suggest calling this "NF core reballancing"

>  - Fix Style guide links

Reviewer: this sounds like a bug fix to me

like `17.11` where the "major" number is the year and the "minor" number
is the month.

## v19.07 (7/19): NFD library and example NFs, Continuous Integration updates, minor improvements and bug fixes.
A CloudLab template is available with the latest release here: https://www.cloudlab.us/p/GWCloudLab/onvm

**Performance**: This release includes a new macro `ENABLE_FLOW_LOOKUP` which controls whether a flow lookup is performed for every incoming packet. If disabled, all packets are forwarded to the default service ID which improves performance. The flow lookup is still enabled by default for backward compatibility with other applications that use ONVM.

### NFD library with example NFS
Add example NFs based on NFD, a C++-based NF developing compiler designed by Wenfei Wu's group (http://wenfei-wu.github.io/) from IIIS, Tsinghua University, China.

> NFD compiles the NF logic into common C++ program by using table-form language to model NFs' behavior . With NFD's help, NF developer, free from the problems that porting NFs into specific network environment or deploying platforms, can easily concentrate on designing NF's core logic. 

Reviewer 2: [nit] into a* common C++ program, extra space after behavior.

A list of provided NFs using NFD library:
 - DNS Amplification Mitigation
 - Super Spread Detection
 - Heavy Hitter Detection
 - SYN Flood Detection
 - UDP Flood Detection
 - Stateless Firewall
 - Stateful Firewall
 - NAPT

### Continuous Integration updates:
CI got a few major updates this release:

>  - CI will do basic lint checks and branch checks(all PRs should be submitted against the *develop* branch) for unathorized users

Reviewer 3: for `unauthorized` users

>  - If CI is working on a request and recives another request it will append it to the queue instead of dropping it

Reviewer 3: and `receives` another
